### PR TITLE
add jenkins-dind docker image and scripts

### PR DESCRIPTION
**Pull request by rji**
For running Jenkins on Mesos, it is preferable to run Docker in Docker
instead of bind-mounting the Docker socket from the host inside the
container. This ensures that we use the resource isolation provided by
Mesos and don't inadvertently circumvent it.

### CODE REVIEW

**Comment by rji** [comment]
Looks like the CI failure was unrelated to this patch:

```
W: Failed to fetch http://httpredir.debian.org/debian/dists/jessie-backports/main/binary-amd64/Packages  Hash Sum mismatch
```
:sob:

File: dind-agent/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+A simple Docker image for running a Jenkins agent alongside its very
+own Docker daemon! Useful if you're trying to run Jenkins agents on a Mesos
+cluster, and you also want to build and push Docker images using your CI system.
+
+## Usage
+### Command line
+Run Jenkins and Docker in Docker with this one neat trick:
+
+```bash
+docker run --privileged rogerignazio/jenkins-dind \
```
Comment (ssk2) [inline dind-agent/README.md]: Update the Docker image name here.

File: dind-agent/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  <executorMem>128</executorMem>
+  <remoteFSRoot>jenkins</remoteFSRoot>
+  <idleTerminationMinutes>3</idleTerminationMinutes>
+  <jvmArgs>
+    -Xms16m -XX:+UseConcMarkSweepGC -Djava.net.preferIPv4Stack=true
+  </jvmArgs>
+  <jnlpArgs/>
+  <containerInfo>
+    <type>DOCKER</type>
+    <dockerImage>rogerignazio/jenkins-dind:0.1</dockerImage>
```
Comment (ssk2) [inline dind-agent/README.md]: Same here.

**Comment by ssk2** [comment]
Can we update the default configuration too?

**Comment by rji** [comment]
Thanks for the review, @ssk2! I just updated the PR with the fixes you mentioned.

**Comment by rji** [comment]
Added a few other required params to the default config.

**Comment by ssk2** [comment]
:ship:

**Pull request merged**